src/main/java/net/daporkchop/fp2/util/Constants.java

```diff
@@ -213,8 +214,7 @@ public static IntAxisAlignedBB getBounds(@NonNull World world) {
             minY = ((ICubicWorld) world).getMinHeight();
             maxY = ((ICubicWorld) world).getMaxHeight() - 1;
         }
-
-        final int HORIZONTAL_LIMIT = 30_000_000; //TODO: hard-coding this is probably a bad idea, but there don't seem to be any variables or methods i can use to get it
+        final int HORIZONTAL_LIMIT = ((IMixinWorldBorder)world.getWorldBorder()).getWorldSize();
```

Comment:
in vanilla, terrain will be rendered up to 30 million blocks even if the world border radius is much smaller (e.g. 100x100). to preserve vanilla behavior when no world border-extending mods are present, this should be the `Math.max` of `WorldBorder#getSize()` and `30_000_000`.